lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -869,8 +886,21 @@ internal class EvaluatingCompiler(
     }
 
     private fun compileSelect(selectExpr: Select): ThunkEnv {
+        // Get all the FROM source aliases for binding error checks
+        val fold = object : PartiqlAst.VisitorFold<Set<String>>() {
+            override fun visitFromSourceScan(node: PartiqlAst.FromSource.Scan, accumulator: Set<String>): Set<String> {
+                val aliases = listOfNotNull(node.asAlias?.text, node.atAlias?.text, node.byAlias?.text)
+                return accumulator + aliases.toSet()
+            }
+
+            override fun walkExprSelect(node: PartiqlAst.Expr.Select, accumulator: Set<String>): Set<String> {
```

Comment:
Add `kdoc` that explains the purpose of this override.

Reply:
Added. Let me know if it doesn't provides sufficient information.

Reply:
Hrm.  I would have expected it to convey that the purpose was to prevent it from recursing into nested select statements.

Reply:
Will change to something like "Prevents visitor from recursing into nested select statements"